src/main/java/core/basesyntax/DataConverterImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : data) {
+            if (line.startsWith("type")) {
+                continue;
+            }
+            String[] parts = line.split(",");
+            FruitTransaction.Operation operation = FruitTransaction.Operation.fromCode(parts[0]);
+            String fruit = parts[1];
+            int quantity = Integer.parseInt(parts[2]);
+            transactions.add(new FruitTransaction(operation, fruit, quantity));
```

Comment:
There is no validation of the input data. Consider adding validation for the parts' length to ensure that there are enough elements to prevent ArrayIndexOutOfBoundsException. Also, consider validating the operation and quantity to ensure they are valid before creating a new FruitTransaction.